net/frr/src/opnsense/scripts/frr/frr_wrapper.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+# Service wrapper for starting/restarting frr service
+# This wrapper is needed to react on specific service interactions through watchfrr.
+# Startup details with watchfrr enabled (default):
+# 1. "service frr start" calls "service frr start watchfrr"
+# 2. watchfrr once started calls "service frr restart all"
+# 3. "restart all" need to loop the list of $frr_daemons to start each
+#    of then
+# 4. vtysh -b is executed to load boot startup configuration
+
+ACTION="$1"
+COMMAND="$2"
+
+/usr/sbin/service frr "$ACTION" "$COMMAND"
+
+# If frr starts/restarts ospfd, e.g. on process error (parameter: start/restart ospfd)
+if [ "$2" = "ospfd" ]; then
+    logger -t frr_wrapper "WATCHFRR - OSPFD - Starting CARP event handler now"
+    /usr/local/opnsense/scripts/frr/carp_event_handler
+fi
+# If frr starts up (parameter: restart all)
+if [ "$2" = "all" ]; then
+    (
+        /usr/bin/logger -t frr_wrapper "WATCHFRR - STARTUP - Starting CARP event handler now"
+        /usr/local/opnsense/scripts/frr/carp_event_handler
+    ) &
+fi
+exit $?
```

Comment:
IMO this is a bit esoteric. If we need to handle return values then better handle them explicitly.